packages/ui/Slider/Slider.js

```diff
@@ -0,0 +1,298 @@
+		 *
+		 * @type {Boolean}
+		 * @public
+		 */
+		pressed: PropTypes.bool,
+
+		/**
+		 * The amount to increment or decrement the value.
+		 *
+		 * @type {Number}
+		 * @default 1
+		 * @public
+		 */
+		step: PropTypes.number,
+
+		/**
+		 * Adds a tooltip to the slider using the provided component.
+		 *
+		 * The following props are forwarded to the tooltip:
+		 *
+		 * * `orientation` - The value of the `orientation` prop from the slider
+		 * * `proportion` - A number between 0 and 1 representing the proportion of the `value` in
+		 *   terms of `min` and `max`
+		 * * `children` - The `value` prop from the slider
+		 *
+		 * This prop accepts either a Component (e.g. `MyTooltip`} which will be instantiated with
+		 * the above props or a component instance (e.g. `<MyTooltip customProp="value" />`) which
+		 * will have its props merged with the above props.
+		 *
+		 * See {@link ui/ComponentOverride} for more information.
+		 *
+		 * @type {Component|Element}
+		 * @public
+		 */
+		tooltipComponent: PropTypes.oneOfType([PropTypes.element, PropTypes.func]),
+
+		/**
+		 * The value of the slider.
+		 *
+		 * @type {Number}
+		 * @default 0
+		 * @public
+		 */
+		value: PropTypes.number
+	},
+
+	defaultProps: {
+		disabled: false,
+		knobComponent: Knob,
+		min: 0,
+		max: 100,
+		noFill: false,
+		orientation: 'horizontal',
+		value: 0
+	},
+
+	styles: {
+		css: componentCss,
+		className: 'slider',
+		publicClassNames: true
+	},
+
+	computed: {
+		className: ({disabled, noFill, orientation, pressed, styler}) => {
+			return styler.append(
+				orientation,
+				{
+					disabled,
+					noFill,
+					pressed
+				}
+			);
+		},
+		percent: ({max, min, value}) => calcProportion(min, max, value),
+		style: ({backgroundProgress, max, min, style, value}) => {
+			const proportion = calcProportion(min, max, value);
+
+			return {
+				...style,
+				'--ui-slider-proportion-end': proportion,
+				'--ui-slider-proportion-end-background': backgroundProgress,
+				'--ui-slider-proportion-end-knob': proportion
+			};
+		}
+	},
+
+	render: ({css, disabled, knobComponent, orientation, percent, tooltipComponent, value, ...rest}) => {
+		delete rest.backgroundProgress;
+		delete rest.max;
+		delete rest.min;
+		delete rest.noFill;
+		delete rest.pressed;
+		delete rest.step;
+
+		return (
+			<div {...rest} disabled={disabled}>
+				<div className={css.bars}>
+					<div className={css.load} />
+					<div className={css.fill} />
+					<ComponentOverride
+						className={css.knob}
+						component={knobComponent}
+						disabled={disabled}
+						orientation={orientation}
+						proportion={percent}
+						tooltipComponent={tooltipComponent}
+						value={value}
+					/>
+				</div>
+			</div>
+		);
+	}
+});
+
+/**
+ * Adds touch and drag support to a [SliderBase]{@link ui/Slider.SliderBase}.
+ *
+ * @hoc
+ * @memberof ui/Slider
+ * @mixes ui/Touchable.Touchable
+ * @public
+ */
+const SliderDecorator = compose(
+	PositionDecorator,
```

Comment:
Hmm ... probably. Unfortunately, we need `Changeable` above `moonstone/Slider` because it mutates the value for 5-way changes. So, the consequence of adding is that we either:

* have an extra layer that isn't adding value, or
* we have to recompose the HOCs in moonstone to avoid the above.

Will experiment with different HOC ordering to avoid both of the above but will likely fall back to the second option.

Reply:
I think I've found a way to include `Changeable` in `ui/SliderDecorator` and then compose `ui/SliderDecorator` in `moonstone/Slider` and `moonstone/IncrementSlider` with minimal extra layers.